spec/helpers/step_nav_actions_helper_spec.rb

```diff
@@ -72,4 +72,35 @@
       expect(helper.can_submit_2i_review?(step_by_step_page, reviewer_user)).to be true
     end
   end
+
+  describe "#can_revert_to_draft?" do
+    let(:step_by_step_page) { create(:step_by_step_page) }
+
+    it "can reverted to draft if it is approved 2i" do
+      step_by_step_page.status = "approved_2i"
+
+      expect(can_revert_to_draft?(step_by_step_page, user)).to be true
+    end
+
+    permitting_statuses = %w[submitted_for_2i in_review]
+
+    permitting_statuses.each do |permitting_status|
+      it "can be reverted to draft if it is #{permitting_status.humanize.downcase} and the user is a review requester" do
+        step_by_step_page.status = permitting_status
+        step_by_step_page.review_requester_id = user.uid
+
+        expect(can_revert_to_draft?(step_by_step_page, user)).to be true
+      end
+    end
+
+    not_permitting_statuses = %w[draft scheduled published]
```

Comment:
```suggestion
    not_permitting_statuses = StepByStepPage::STATUSES - permitting_statuses
```

Reply:
I don't think this will work because `approved_2i` in not in `permitting_statuses`

Reply:
Then maybe this?

```suggestion
    not_permitting_statuses = StepByStepPage::STATUSES - permitting_statuses - "approved_2i"
```

Reply:
Perhaps:
```
allowed_statuses = %w[submitted_for_2i in_review approved_2i]
user_restricted_statuses = allowed_statuses - "approved_2i"
disallowed_statuses = StepByStepPage::STATUSES - allowed_statuses
```

I'm not sure if that makes anything clearer of not.